plugins/social_network/social_network.cpp

```diff
@@ -374,6 +464,9 @@ namespace golos { namespace plugins { namespace social_network {
             ) (
                 "set-content-storing-depth-null-after-update", boost::program_options::value<bool>()->default_value(false),
                 "should content's depth be set to null after update"
+            ) (
+                "comment-last-update-depth", boost::program_options::value<uint32_t>()->default_value(std::numeric_limits<uint32_t>::max()),
+                "mode of storing records of comment.active and comment.last_update: -1 = store all, 0 = do not store, N = storing N blocks depth"
```

Comment:
It is bad to write about `-1`.
For some compilation flags  it can be not work.
Fix comment `-1` -> `4294967295`

Reply:
@afalaleev Isn't so big number quite inconvenient for users?
-1 is frequently used as a value for "no count", for instance in WinAPI.
Why not use a signed int and `default_value(-1)`?

Reply:
`-1` doesn't work in all cases.
There will be questions from users who will set value `-1` but it doesn't work.

Reply:
@afalaleev All will work and be simple if use `int32_t` without `uint32_t`, but ok

Reply:
block_number has type `uint32_t`

Reply:
sorry to interrupt, but I have similar code in my vote limiting implementation. `-1` should not be a problem here, because it casted to `uint32_t` and become big value. It's big enough to skip explicit check of -1 or 0xFFFFFFFF, so there no need to separate logic, it will be enough for store all.
furthermore, some api plugins already use same pattern (use -1 with `uint32_t`) and it works fine. am I missing something?

Reply:
If we check on some big value, there is no matter. 
But in this code we use a magic around constant, and it may not work.
To fix we may add two checks on `-1` and `std::numeric_limits<uint32_t>::max()`